### PR TITLE
GCP resources API

### PR DESCRIPTION
**Pull request by tjholm**
Add Details API for GCP resources.

Currently only implements APIs as requested.

### CODE REVIEW

File: pkg/providers/gcp/core/provider.go [hunk 1]
```diff
@@ -53,6 +64,49 @@ func createMetadataRequest(uri string) (*http.Request, error) {
...
+			Detail: common.ApiDetails{
+				URL: fmt.Sprintf("https://%s", gw.DefaultHostname),
+			},
+		}, nil
+	} else {
+		return nil, err
+	}
+}
+
+func (g *gcpProviderImpl) Details(typ common.ResourceType, name string) (*common.DetailsResponse[any], error) {
```
Comment (HomelessDinosaur) [inline pkg/providers/gcp/core/provider.go:101]:
```suggestion
func (g *gcpProviderImpl) Details(type common.ResourceType, name string) (*common.DetailsResponse[any], error) {
```

File: pkg/providers/gcp/core/provider.go [hunk 2]
```diff
@@ -53,6 +64,49 @@ func createMetadataRequest(uri string) (*http.Request, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	gws := g.apiClient.ListGateways(context.TODO(), &apigatewaypb.ListGatewaysRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/%s", projectName, g.region),
+		Filter: filter(g.stackName, name),
+	})
+
+	// there should only be a single entry in this array, we'll grab the first and then brake
```
Comment (HomelessDinosaur) [inline pkg/providers/gcp/core/provider.go:86]:
```suggestion
	// there should only be a single entry in this array, we'll grab the first and then break
```

**Review by HomelessDinosaur — CHANGES_REQUESTED** [review]

File: pkg/providers/gcp/core/provider.go (hunk 1 above)
Comment (tjholm) [reply pkg/providers/gcp/core/provider.go:101]: `type` is a reserved keyword in golang

**Review by davemooreuws — APPROVED** [review]
LGTM

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**